roles/foreman_installer/tasks/main.yml

```
@@ -0,0 +1,29 @@
+---
+- name: Check that necessary variables are defined
+  assert:
+    that:
+      - foreman_installer_scenario is defined
+
+- name: Check if initial install has happened
+  find:
+    paths: /etc/foreman-installer/scenarios.d
+    patterns: .installed
+  register: installed
+
+- name: 'Run installer in noop mode'
+  command: >
+    {{ foreman_installer_command }}
+    --scenario {{ foreman_installer_scenario }}
+    --noop
+  register: foreman_installer_noop
+  changed_when: false
+  when: installed.matched == 1
+
+- name: 'Run installer' # noqa 301
+  command: >
+    {{ foreman_installer_command }}
+    --scenario {{ foreman_installer_scenario }}
+    {{ (foreman_installer_verbose|bool) | ternary("-v", "") }}
+    {{ (foreman_installer_no_colors|bool) | ternary("--no-colors", "") }}
+    {{ foreman_installer_options | join(' ') }}
+  when: not foreman_installer_noop.skipped and foreman_installer_noop.rc != 0
```

```suggestion
  when:
    - not foreman_installer_noop.skipped 
    - foreman_installer_noop.rc != 0
```

Doesn't this mean the role never does an initial install, as this is missing the `.installed` file?